worker/uniter/charm/bundles.go

```diff
@@ -122,8 +133,29 @@ func (d *BundlesDir) bundleURLPath(url *charm.URL) string {
 	return path.Join(d.path, charm.Quote(url.String()))
 }
 
+// ClearDownloads removes any entries in the temporary bundle download
+// directory. It is intended to be called on uniter startup.
+func ClearDownloads(bundlesDir string) (err error) {
+	defer errors.DeferredAnnotatef(&err, "unable to clear bundle downloads")
+
+	downloadDir := downloadsPath(bundlesDir)
+	entries, err := ioutil.ReadDir(downloadDir)
+	if os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return
+	}
+	for _, entry := range entries {
+		path := filepath.Join(downloadDir, entry.Name())
+		if err = os.RemoveAll(path); err != nil {
+			return
```

Comment:
+1, though I'm not sure how helpful it'll be in practice. Assuming it's a transient error, the worker should restart and eventually clean them all up.